packages/advanced-logic/specs/payment-network-erc20-escrow-0.1.0.md

```diff
@@ -0,0 +1,55 @@
+# Additional interpretation for ERC20 Fee Proxy: smart escrow (WIP)
+
+**TODO: These specifications are work in progress and might change a lot before they are fully supported by the library.**
+
+Prerequisite: Payment Network "ERC20 With Fee" (see [here](./payment-network-erc20-fee-proxy-contract-0.1.0.md)).
+
+## Description
+
+This is an additional interpretation for the payment network "ERC20 With Fee", allowing the ERC20 payments to be locked on an escrow contract.
+
+The payment is made beforehand to the smart escrow, which locks the funds. The payer can then unlock the funds for a certain target date, after which the request issuer can withdraw the funds. In case of collaboration conflict, the payer can also decide to freeze the payment, which triggers a one-year period freeze. After a freeze, only the payer can withdraw the funds.
+
+The escrow contract pulls ERC20 tokens from the user, there should be no manual transfer.
+
+The escrow contract relies on a proxy when funds are withdrawn. The proxy is responsible for sending `TransferWithReferenceAndFee` events and process the payment to the issuer.
```

Comment:
can the proxy be administrated?

Reply:
Two way of thinking this:
- administrated make easier maintenance (avoid to redeploy new smartcontract when the proxy need to be modify)
- administrated mean less decentralization (people need to trust Request to not sneaky change the proxy to a malicious one)

Reply:
I think the low frequency of proxy updates (especially: with the same function signature) makes it less relevant. We've never used it for Swap.